tests/android/androidTest.ts

```diff
@@ -65,4 +65,24 @@ export const androidTest = baseTest.extend<PageTestFixtures, AndroidWorkerFixtur
     const page = await androidContext.newPage();
     await run(page);
   },
+
+  rewriteRequestToLocalhost: [async ({ isAndroid, rewriteRequestURLToLocalhost }, use) => {
+    await use(request => {
+      if (!isAndroid)
+        return request;
+      const initializer = (request as any)._initializer;
+      initializer.url = rewriteRequestURLToLocalhost(initializer.url);
+      return request;
+    });
+  }, { scope: 'worker' }],
+
+  rewriteRequestURLToLocalhost: [async ({ isAndroid }, use) => {
```

Comment:
Let's make this a local fixture `rewriteAndroidLoopbackURL` in the interception test file and add comments:
```js
// We access test servers at 10.0.2.2 from inside the browser on Android,
// which is actually forwarded to the desktop localhost.
// To use request such an url with apiRequestContext on the desktop, we need to change it back to localhost.
```